src/python/dxpy/bindings/__init__.py

```diff
@@ -657,6 +657,33 @@ def _wait_on_close(self, timeout=3600*24*1, **kwargs):
             i += 1
             elapsed += wait
 
+    def _wait_until_parts_uploaded(self, timeout=60, **kwargs):
+        elapsed = 0
+        i = 0
+        while True:
+            parts = self.describe(fields={'parts'}, **kwargs)["parts"]
+            state = self._get_state(**kwargs)
+            if state == "closed":
+                # parts of closed files must have been uploaded successfully
+                break
+            if not parts:
+                break
+            else:
+                is_uploaded = True
+                for key in parts:
+                    if parts[key].get(state, 'complete') != 'complete':
+                        is_uploaded = False
+                        break
+                if is_uploaded:
+                    break
+            if elapsed >= timeout or elapsed < 0:
+                raise DXError("Reached timeout while waiting for parts of the file ({}) to be uploaded".format(self.get_id()))
+
+            wait = min(2**7, 2**i)
```

Comment:
Consider having n=11 as opposed n=7 for this.  This an important timeout because it's a global limit on the upload file (1+2+4+8+... geometric series). Setting n=11 will result in this closed form sum for the geometric series: `((1-2^(11+1))/(1-2))/60` which is an hour and change.  That seems reasonable, but I feel like we need real data to base this on before landing it.  We may want to even expose this parameter for cases of really large files.  I will chat with the data team on this more.

Reply:
Sorry for my misunderstanding.  Yes, a smaller timeout makes sense perhaps setting `timeout` to  something like `((1-2^(7+1))/(1-2))` so it will allow all 7 retries to occur.